examples/plot_int_alignment.py

```diff
@@ -0,0 +1,256 @@
+#   - target train: sub-07 AP contrasts, used to learn alignment to template
+#   - target test: sub-07 PA contrasts, used as a ground truth to score predictions
+# We make a single 4D Niimg from our list of 3D filenames
+
+target_train = concat_imgs(target_train)
+target_train_data = masker.transform(target_train)
+target_test = df[df.subject == "sub-07"][df.acquisition == "pa"].path.values
+
+
+###############################################################################
+# Compute a baseline (average of subjects)
+# ----------------------------------------
+# We create an image with as many contrasts as any subject representing for
+# each contrast the average of all train subjects maps.
+#
+
+import numpy as np
+
+masked_imgs = [masker.transform(img) for img in template_train]
+average_img = np.mean(masked_imgs[:-1], axis=0)
+average_subject = masker.inverse_transform(average_img)
+
+###############################################################################
+# Create a template from the training subjects.
+# ---------------------------------------------
+# We define an estimator using the class TemplateAlignment:
+#   * We align the whole brain through 'multiple' local alignments.
+#   * These alignments are calculated on a parcellation of the brain in 150 pieces,
+#     this parcellation creates group of functionnally similar voxels.
+#   * The template is created iteratively, aligning all subjects data into a
+#     common space, from which the template is inferred and aligning again to this
+#     new template space.
+#
+
+from nilearn.image import index_img
+from fmralign.alignment_methods import IndividualizedNeuralTuning
+from fmralign.hyperalignment.regions import compute_parcels, compute_searchlights
+
+###############################################################################
+# Predict new data for left-out subject
+# -------------------------------------
+# We use target_train data to fit the transform, indicating it corresponds to
+# the contrasts indexed by train_index and predict from this learnt alignment
+# contrasts corresponding to template test_index numbers.
+# For each train subject and for the template, the AP contrasts are sorted from
+# 0, to 53, and then the PA contrasts from 53 to 106.
+#
+
+train_index = range(53)
+test_index = range(53, 106)
+full_index = range(106)
+
+train_data = np.array(masked_imgs)[:, train_index, :]
+test_data = np.array(masked_imgs)[:, full_index, :]
+target_test_masked = np.array(masked_imgs)[:, test_index, :]
+
+if False:
+    parcels = compute_parcels(
+        niimg=template_train[0], mask=masker, n_parcels=100, n_jobs=5
+    )
+    model = IndividualizedNeuralTuning(
+        n_jobs=8, alignment_method="parcellation", n_components=None
+    )
+    model.fit(train_data, parcels=parcels, verbose=False)
+    train_stimulus = np.copy(model.shared_response)
+    train_tuning = np.linalg.pinv(train_stimulus) @ model.denoised_signal[-1]
+    model_bis = IndividualizedNeuralTuning(
```

Comment:
It doe snot make sense to instantiate 2 INTs. By doing so, you're creating two representations without correspondence.

Reply:
@bthirion This is all the problem with INT : at some point stimulus matrices will loose correspondance. Even if we do co-smoothing with 'closing the square', thus synchronizing stimulus between train and test, there is still no guarantee that the reconstructed image will be in correspondance with the unseen data, as even with a synchronised stimulus, there is still a loss in the order of sources compared to ground truth signal (because we have too cut the stimulus matrix in half at some point to extract the training tuning matrix for the left out subject)